streams: Add notifications for posting policy changes.

eeshangarg opened the pull request:
Here is a screenshot of a sample notification:
![stream_post_policy_notification](https://user-images.githubusercontent.com/7251823/148621639-784f0a31-c175-4647-93fe-5d656d6e8dbf.png)

@timabbott: I split out the stream posting policy changes out of #20540 into a smaller PR, with the following new changes:

* We now use `STREAM_PROPERTY_CHANGED` with the property name encoded in the `extra_data` dictionary in our `RealmAuditLog` entries.
* Made `acting_user` a required keyword argument in `do_change_stream_post_policy`.
* Changed the `user_profile` argument to `acting_user` in `send_change_stream_post_policy_notification`, for consistency with the calling code.
* Also modified the test in `test_events.py` to use `check_message` to verify the new event.

@alya This is the change where Tim was asking for a potential revision to the wording of the notification message. See [this comment by Tim](https://github.com/zulip/zulip/pull/20540#discussion_r779234225). Thanks! :)

alya commented:
Two options that come to mind:

1) Use the text above, but with a colon: "... for this stream: only organization administrators..." 
2) Use a similar structure to stream descriptions changes:

{user} changed the [posting permissions](/help/stream-sending-policy).

- **Old permissions:** ...
- **New permissions:** ...

Thoughts?

timabbott commented:
Following the stream description format might feel more readable. If we don't do that, we might want to have it be `Now, only ...`.

eeshangarg commented:
@timabbott: This PR has now been updated with the following changes:
* I fixed the `tools/test-api` failures earlier today. I also added an explanatory note in the commit description describing the rationale behind my changes.
* I also changed the notification message to use a similar template to what we use for description changes. See updated screenshot below (@alya FYI):

![posting_permissions_notification](https://user-images.githubusercontent.com/7251823/148850486-31b51544-3ea9-468b-8cf6-8f448686e156.png)

This PR is now ready for another review! Thanks!

eeshangarg commented:
@timabbott Just updated this PR based on the feedback on CZO in [this thread](https://chat.zulip.org/#narrow/stream/101-design/topic/stream.20settings.20notifications/near/1307705).

Latest screenshot:
![stream_post_policy_notification](https://user-images.githubusercontent.com/7251823/148861821-cdc173c0-1f30-458b-ad9d-046ddf0e1a89.png)

timabbott commented on zerver/models.py:
```diff
@@ -2253,12 +2253,17 @@ class Stream(models.Model):
-        STREAM_POST_POLICY_RESTRICT_NEW_MEMBERS,
-        STREAM_POST_POLICY_MODERATORS,
-    ]
+    POST_POLICIES: Dict[int, str] = {
```
I added a comment here and in the frontend about keeping these strings matching.

timabbott commented:
Merged, after adding a comment about keeping the string descriptions the same.
